graph/src/components/store.rs

```diff
@@ -1478,13 +1489,19 @@ impl EntityCache {
                     // Merging with an empty entity removes null fields.
                     let mut data = Entity::new();
                     data.merge_remove_null_fields(updates);
+                    if let Some(fulltext_fields) = fulltext_entity_fields.get(&key.entity_type) {
```

Comment:
I don't agree, then we are passing an Entity into the store that is partially updated, but still contains the old value for any fulltext fields.  

I explored how to move this update into `apply_entity_modifications()` or `InsertQuery::new()` , but I don't see how that makes things simpler.  To make the changes in InsertQuery::new I'd need access to a reference of which fields are included in fulltext fields, so that would mean adding a HashSet of fulltext fields to Column or a reference of all fulltext fields to their included fields on the Table.  The other issue is that we are passing in a reference to the Entity that will be applied, would we then need to pass in a mutable reference? It feels much cleaner to me to update the EntityModifications all at the same time, still pass in a reference to the Entity and not require passing fulltext field references down into the query execution.